src/lib.rs

```diff
@@ -133,6 +144,11 @@ pub trait LanguageServer: Send + Sync + 'static {
     type TypeDefinitionFuture: Future<Item = Option<GotoDefinitionResponse>, Error = Error> + Send;
     /// Response returned when a document highlight action is requested.
     type HighlightFuture: Future<Item = Option<Vec<DocumentHighlight>>, Error = Error> + Send;
+    /// Response returned when a signature help action is requested.
+    type SignatureHelpFuture: Future<Item = Option<SignatureHelp>, Error = Error> + Send;
+    /// Response returned when a goto implementation action is requested.
+    type GotoImplementationFuture: Future<Item = Option<GotoImplementationResponse>, Error = Error>
```

Comment:
This future should be renamed to `ImplementationFuture` to be consistent with `TypeDefinitionFuture` and moved to immediately below `TypeDefinitionFuture`.